autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/CourseSelectCard.tsx

```diff
@@ -1,35 +1,202 @@
+        </div>
+        <div style={{ display: showSection ? 'contents' : 'none' }}>
+          <SectionSelect id={id} />
+        </div>
+        <div style={{ display: showHint ? 'block' : 'none' }}>
+          <Typography color="textSecondary">
+            Select a course to show available sections
+          </Typography>
+        </div>
+      </>
+    );
+  };
+
+  const collapsibleContent = (
+    <div className={styles.content} aria-hidden={collapsed}>
+      <Autocomplete
+        options={options}
+        size="small"
+        autoHighlight
+        autoSelect
+        disabled={loading}
+        inputValue={inputValue}
+        value={course}
+        multiple={false}
+        filterOptions={(): any[] => options} // Options are not filtered
+        getOptionSelected={(option): boolean => option === options[0]}
+        onClose={(): void => {
+          if (options.length === 0) setInputValue('');
+        }}
+        onChange={(_evt: object, val: string): void => {
+          dispatch(updateCourseCard(id, {
+            course: val,
+          }, term));
+        }}
+        onInputChange={(_evt: object, val: string, reason: string): void => {
+          setInputValue(val);
+          if (val === '') { // Skip empty inputs
+            setOptions([]); // Clear the autocomplete options so previous results don't show
+            return;
+          }
+
+          if (reason !== 'reset') getAutocomplete(val);
+        }}
+        renderInput={(params: any): JSX.Element => (
+          // eslint-disable-next-line react/jsx-props-no-spreading
+          <TextField {...params} label="Course" fullWidth autoFocus variant="outlined" />
+        )}
+        classes={{ root: styles.courseInput }}
+      />
+      <FormLabel component="label" style={{ marginTop: 16 }} focused={false}>
+          Customization Level:
+      </FormLabel>
+      <ButtonGroup className={styles.customizationButtons}>
+        <Button
+          className={styles.noElevation}
+          color="primary"
+          variant={customizationLevel === CustomizationLevel.BASIC ? 'contained' : 'outlined'}
+          onClick={(): void => {
+            dispatch(updateCourseCard(id, {
+              customizationLevel: CustomizationLevel.BASIC,
+            }));
+          }}
+        >
+            Basic
+        </Button>
+        <Button
+          className={styles.noElevation}
+          color="primary"
+          variant={customizationLevel === CustomizationLevel.SECTION ? 'contained' : 'outlined'}
+          onClick={(): void => {
+            dispatch(updateCourseCard(id, {
+              customizationLevel: CustomizationLevel.SECTION,
+            }));
+          }}
+        >
+            Section
+        </Button>
+      </ButtonGroup>
+      {getCustomizationContent()}
+    </div>
+  );
+
+  return (
+    <Card className={styles.card}>
+      {header}
+      <Collapse in={!collapsed} appear enter={shouldAnimate} onEntered={resetAnimCb}>
```

Comment:
what does `enter` do? I can't find any mention of it in the docs

Reply:
Material UI internally uses React Transition Group, and the `enter` prop gets forwarded to it. See [docs](https://reactcommunity.org/react-transition-group/transition)